src/ctelnet.cpp

```diff
@@ -1603,27 +1597,23 @@ int cTelnet::decompressBuffer( char *& in_buffer, int& length, char* out_buffer
         inflateEnd( & mZstream );
         qDebug() << "recv Z_STREAM_END, ending compression";
         this->mNeedDecompression = false;
-        this->mMCCP_version_1 = false;
-        this->mMCCP_version_2 = false;
-
-        // Reset the option state so we can re-enable compression again in the future
-        // such as in the case of a copyover -JM
-        hisOptionState[static_cast<int>(OPT_COMPRESS)] = false;
-        hisOptionState[static_cast<int>(OPT_COMPRESS2)] = false;
```

Comment:
I think you do need to keep those two `hisOptionState` up to date because they would be needed for a Telnet Status sub-option report _which we DO support_ (see code for `OPT_STATUS` i.e. option **5**).